src/datatypes/datetime.coffee

```diff
@@ -841,6 +841,10 @@ DateTime.prototype.add = Date.prototype.add = (offset, field) ->
     for f in @constructor.FIELDS
       result[f] = null if not @[f]?
 
+  # To combat the conversion of the year 0 to 1900
```

Comment:
What is the conversion of the year 0 to 1900?

Reply:
When the date gets converted to a javascript Date object in the toJSDate() function on line 267, if the year is equal to 0, it returns a year of 1900. This affected when the min_date is used.

Reply:
Interesting--it looks like JS Date behaves differently for years 0-100, assuming they are in the 1900s.  It seems like there is a way to use setUTCFullYear to adjust for this 'feature'.  I think we should do that and move it to the toJSDateFunction rather than having it here.